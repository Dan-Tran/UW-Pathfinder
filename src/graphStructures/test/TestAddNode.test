# This will test adding nodes to the graph.

# Create the graph.
CreateGraph graphtest

# Make sure the graph has no nodes and no edges.
ListNodes graphtest

# Add one node
AddNode graphtest n1

# Make sure the node was added
ListNodes graphtest

# Make sure there is no edges
ListChildren graphtest n1

# Add another node
AddNode graphtest n2

# Make sure the node was added
ListNodes graphtest

#Makesure there is no edges
ListChildren graphtest n1
ListChildren graphtest n2

# Add multiple nodes.
AddNode graphtest n3
AddNode graphtest n4
AddNode graphtest n5

# Make sure the nodes were added
ListNodes graphtest

# Make sure there are no edges
ListChildren graphtest n1
ListChildren graphtest n2
ListChildren graphtest n3
ListChildren graphtest n4
ListChildren graphtest n5