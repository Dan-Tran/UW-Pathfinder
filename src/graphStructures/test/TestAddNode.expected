# This will test adding nodes to the graph.

# Create the graph.
created graph graphtest

# Make sure the graph has no nodes and no edges.
graphtest contains:

# Add one node
added node n1 to graphtest

# Make sure the node was added
graphtest contains: n1

# Make sure there is no edges
the children of n1 in graphtest are:

# Add another node
added node n2 to graphtest

# Make sure the node was added
graphtest contains: n1 n2

#Makesure there is no edges
the children of n1 in graphtest are:
the children of n2 in graphtest are:

# Add multiple nodes.
added node n3 to graphtest
added node n4 to graphtest
added node n5 to graphtest

# Make sure the nodes were added
graphtest contains: n1 n2 n3 n4 n5

# Make sure there are no edges
the children of n1 in graphtest are:
the children of n2 in graphtest are:
the children of n3 in graphtest are:
the children of n4 in graphtest are:
the children of n5 in graphtest are: